File: packages/sdk/android/src/main/java/com/primerioreactnative/components/events/PrimerHeadlessUniversalCheckoutComponentEvent.kt
package com.primerioreactnative.components.events

internal enum class PrimerHeadlessUniversalCheckoutComponentEvent(val eventName: String) {
    ON_STEP("onStep"),
    ON_ERROR("onError"),
    ON_IN_VALID("onInvalid"),
    ON_VALID("onValid"),
    ON_VALIDATING("onValidating"),
    ON_VALIDATION_ERROR("onValidationError"),
}

File: packages/sdk/ios/Sources/DataModels/Klarna Component/PrimerHeadlessUniversalCheckoutComponentEvent.swift
//
//  PrimerHeadlessUniversalCheckoutComponentEvent.swift
//  primer-io-react-native
//
//  Created by Stefan Vrancianu on 01.04.2024.
//

import Foundation

// swiftlint:disable type_name
@objc
enum PrimerHeadlessUniversalCheckoutComponentEvent: Int, CaseIterable {
    // swiftlint:enable type_name

    case onStep
    case onError
    case onValid
    case onInvalid
    case onValidating
    case onValidationError

    var stringValue: String {
        switch self {
        case .onStep:
            return "onStep"
        case .onError:
            return "onError"
        case .onValid:
            return "onValid"
        case .onInvalid:
            return "onInvalid"
        case .onValidating:
            return "onValidating"
        case .onValidationError:
            return "onValidationError"
        }
    }
}
